lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -133,9 +148,11 @@ newTransactionLayer (Hash block0H) = TransactionLayer
         wits <- forM rnps $ \(_, TxOut addr _) -> do
             xprv <- maybeToRight (ErrKeyNotFoundForAddress addr) (keyFrom addr)
             let body = block0H <> getHash (signData payload inps outs)
-            pure $ mkTxWitness (fst xprv) (sign body xprv)
+            pure $ mkTxWitness (fst xprv) (sign xprv body)
         let tx = Tx
-                { txId = fragmentId msgType payload inps outs wits
+                { txId = error "we shouldn't try to calculate this yet"
```

Comment:
I bet this is temporary and to be corrected in coming commits

Reply:
Yes – this is what makes me believe that we have to do something radically different.

I'm looking at completely diverging from this PR, and doing some refactoring first.